### PR TITLE
Induce component version using ldflags

### PR DESCRIPTION
**Pull request by shafeeqes**
**How to categorize this PR?**

/area networking
/kind enhancement

**What this PR does / why we need it**:
Currently, it is hard to find the version of the `networking-cilium` images.
This PR improves this by inducing the version using ldflags.
Version can be printed by using the `--version` flag.

Also we log the version during the start of the component.

Also with this PR, docker images built by make docker-images are tagged and built with a version including the commit hash (exactly like the images built by the pipeline).

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Similar to https://github.com/gardener/gardener-extension-networking-calico/pull/266

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,10 +20,15 @@ IMAGE_PREFIX                := $(REGISTRY)/extensions
 REPO_ROOT                   := $(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
 HACK_DIR                    := $(REPO_ROOT)/hack
 VERSION                     := $(shell cat "$(REPO_ROOT)/VERSION")
-LD_FLAGS                    := "-w -X github.com/gardener/$(EXTENSION_PREFIX)-$(NAME)/pkg/version.Version=$(IMAGE_TAG)"
+EFFECTIVE_VERSION           := $(VERSION)-$(shell git rev-parse HEAD)
+LD_FLAGS := "-w $(shell $(REPO_ROOT)/vendor/github.com/gardener/gardener/hack/get-build-ld-flags.sh "" $(REPO_ROOT)/VERSION "$(EXTENSION_PREFIX)")"
```
Comment (ScheererJ) [inline Makefile:24]: Please align as before.

```suggestion
LD_FLAGS                    := "-w $(shell $(REPO_ROOT)/vendor/github.com/gardener/gardener/hack/get-build-ld-flags.sh "" $(REPO_ROOT)/VERSION "$(EXTENSION_PREFIX)")"
```

File: cmd/gardener-extension-networking-cilium/app/app.go [hunk 2]
```diff
@@ -40,6 +44,9 @@ import (
 	"github.com/gardener/gardener-extension-networking-cilium/pkg/healthcheck"
 )
 
+// Name is a constant for the name of this component.
+const Name = "gardener-extension-networking-calico"
```
Comment (ScheererJ) [inline cmd/gardener-extension-networking-cilium/app/app.go:48]: Please correct the name.

```suggestion
const Name = "gardener-extension-networking-cilium"
```

**Review by ScheererJ — CHANGES_REQUESTED** [review]
Please also adapt the PR description and replace calico with cilium where necessary.

**Comment by shafeeqes** [comment]
@ScheererJ Can we merge this?

**Pull request merged**